# Data preparation

## Script for precip validation

This script checks the validity of the precip column, whether it contains any numbers other than 0 and 1.

In [2]:
import pandas as pd

df_weather_2019 = pd.read_csv("weather_2019.csv", sep=",")
df_weather_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8194 entries, 0 to 8193
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  8194 non-null   int64  
 1   date_time   8194 non-null   object 
 2   max_temp    8194 non-null   float64
 3   min_temp    8194 non-null   float64
 4   precip      8194 non-null   float64
dtypes: float64(3), int64(1), object(1)
memory usage: 320.2+ KB


The precip column contains 8194 entries, so the number of 0 and 1 in total should equal to 8194. Let's check:

In [3]:
a = df_weather_2019.precip[df_weather_2019.precip == 0].count()
b = df_weather_2019.precip[df_weather_2019.precip == 1].count()
print(a + b)

8194


Done!